### PR TITLE
evm: remove isTruthy and isFalsy

### PR DESCRIPTION
**Pull request by gabrocheleau**
Extracts evm from #2233

### CODE REVIEW

File: packages/evm/src/evm.ts [hunk 1]
```diff
@@ -577,7 +575,7 @@ export class EVM implements EVMInterface {
         if (this.DEBUG) {
           debug(`Not enough gas or code size not allowed (>= Homestead)`)
         }
-        result = { ...result, ...CodesizeExceedsMaximumError(message.gasLimit) }
+        result = { ...result, ...OOGResult(message.gasLimit) }
```
Comment (jochem-brouwer) [inline packages/evm/src/evm.ts:578]: Why is this removed? The error message is handy, because then you know the reason why the contract could not be created.

**Review by jochem-brouwer — COMMENTED** [review]
One Q

File: packages/evm/src/evm.ts (hunk 1 above)
Comment (gabrocheleau) [reply packages/evm/src/evm.ts:578]: Hmm strange, I don't remember changing that. Will take a deeper look

Comment (jochem-brouwer) [reply packages/evm/src/evm.ts:578]: How come this test is not failing? https://github.com/ethereumjs/ethereumjs-monorepo/blob/f1cf6b31e6f1be3c6f71a1b8fe0696fb23cedf13/packages/evm/tests/runCall.spec.ts#L574

Comment (gabrocheleau) [reply packages/evm/src/evm.ts:578]: Ah now I see what happened, I extracted these commits with my branch not updated with that commit: https://github.com/ethereumjs/ethereumjs-monorepo/commit/de21d490e60999f667733625a49cc72dc23f21b8 so these changes were missed

Comment (gabrocheleau) [reply packages/evm/src/evm.ts:578]: Hmm but yeah, not sure why the test wasn't failing?

Comment (jochem-brouwer) [reply packages/evm/src/evm.ts]: I checked locally, it failed for me, and the CI should also pick it up (`npm run coverage`)

**Review by acolytec3 — APPROVED** [review]
LGTM

File: packages/evm/src/precompiles/0e-bls12-g2mul.ts [hunk 2]
```diff
@@ -13,8 +11,6 @@ const {
 } = require('./util/bls12_381')
 
 export async function precompile0e(opts: PrecompileInput): Promise<ExecResult> {
-  if (isFalsy(opts.data)) throw new Error('opts.data missing but required')
```
Comment (holgerd77) [inline packages/evm/src/precompiles/0e-bls12-g2mul.ts:16]: Hmm, I already overlooked a similar case on some other PR, but since it is reoccuring here: I vaguely remember that we introduced this for the JS side of things, since there error messages are unreliable if some parameter is not passed in correctly and things are not guarded by type checking.

Not sure, eventually worth to re-add.

On the other hand, to admit, we do not really have a consistent JS guarding strategy throughout the monorepo anyhow.

Hmm.

Anhow: just wanted to at least mention.
